Feature/contract store

strider8796 opened the pull request:
Branch 정보에 포함된(branch-yggdrash.json) contractVersion.jar 파일이 존재하지 않을 경우,
AWS S3에서 contractVersion.jar 파일을 다운 받아서 osgi 기동시키는 로직 추가했습니다.

yggdrash.conf에 s3주소 추가해주셔야합니다.
```
contract {
  # Contract class file path (default: '$WORKING_DIR$/.yggdrash/contract')
  path = ${YGG_DATA_PATH}"/contract"
  url = "https://s3.ap-northeast-2.amazonaws.com/store.yggdrash.io/contract/"
}
```

Commit:
Implement contract downloader

need to add blow configuration in yggdrash.conf file
contract {
  # ...
  url = "https://s3.ap-northeast-2.amazonaws.com/store.yggdrash.io/contract/"
}

Commit:
Implement contract downloader: missing configuration

need to add blow configuration in yggdrash.conf file
contract {
  # ...
  url = "https://s3.ap-northeast-2.amazonaws.com/store.yggdrash.io/contract/"
}

Commit:
Fix business logic error

in previous version, did not install to osgi that contract file that already exist

RachaelHong commented:
Download 해보니 잘 되네요~ 그런데 Contract class file path 에 contract 디렉토리가 존재하지 않으면 

Downloading contractFile 178b44b22d8c6d5bb08175fa2fcab15122ca8d1e has an error occurred.
Verifying contract file 178b44b22d8c6d5bb08175fa2fcab15122ca8d1e has an error occurred.

로그가 찍히는데, 로그에 해당 내용을 찍어주는건 어떤가요? contract directory 가 존재 하지 않아서인지, 실제 download 시 문제 인지 알아야 하지 않을까 해서요~

Commit:
Update contract downloader

recomposing download flow
make contract directory when if directory does not exist
delete contract file if download failed or verify failed

Commit:
Hot-fix Update contract downloader

change wrong flag
